File: src/main/java/io/vproxy/vfx/theme/impl/DarkThemeFontProvider.java
package io.vproxy.vfx.theme.impl;

import io.vproxy.vfx.manager.font.FontManager;
import io.vproxy.vfx.manager.font.FontSettings;
import io.vproxy.vfx.manager.font.FontUsage;
import io.vproxy.vfx.theme.ThemeFontProvider;

public class DarkThemeFontProvider extends ThemeFontProvider {
    protected void defaultFont(FontSettings settings) {
        settings.setFamily(FontManager.FONT_NAME_NotoSansSCRegular);
    }

    @Override
    protected void windowTitle(FontSettings settings) {
        defaultFont(settings);
        settings.setSize(15);
    }

    @Override
    protected void tableCellText(FontSettings settings) {
        defaultFont(settings);
        settings.setSize(12);
    }

    @Override
    protected void _default(FontUsage usage, FontSettings settings) {
        defaultFont(settings);
        settings.setSize(16);
    }
}

File: src-ts/src/theme/impl/DarkThemeFontProvider.ts
import { FontManager } from '../../manager/font/FontManager.js';
import type { FontSettings } from '../../manager/font/FontSettings.js';
import type { FontUsage } from '../../manager/font/FontUsage.js';
import { ThemeFontProvider } from '../ThemeFontProvider.js';

export class DarkThemeFontProvider extends ThemeFontProvider {
  protected defaultFont(settings: FontSettings): void {
    settings.setFamily(FontManager.FONT_NAME_NotoSansSCRegular);
  }

  protected windowTitle(settings: FontSettings): void {
    this.defaultFont(settings);
    settings.setSize(15);
  }

  protected tableCellText(settings: FontSettings): void {
    this.defaultFont(settings);
    settings.setSize(12);
  }

  protected _default(_usage: FontUsage, settings: FontSettings): void {
    this.defaultFont(settings);
    settings.setSize(16);
  }
}
